DataStructures/Dictionaries/OpenAddressingHashTable.cs

```diff
@@ -0,0 +1,395 @@
+﻿using System;
```

Comment:
This is a new data structure implementation, correct? Can you maybe take this out of this PR and submit it as a separate one, this PR is kind of bloated, it has fixes for the warnings as well as many other things, making this new contribution to the code kind of lost in a huge PR.